petl/io/gsheet.py

```diff
@@ -0,0 +1,156 @@
+            ws = wb.worksheet(text_type(self.worksheet_title))
+
+        # grab the range or grab the whole sheet
+        if self.range_string:
+            # start_cell -> top left, end_cell -> bottom right
+            start_cell, end_cell = self.range_string.split(':')
+            start_row, start_col = gspread.utils.a1_to_rowcol(start_cell)
+            end_row, end_col = gspread.utils.a1_to_rowcol(end_cell)
+            # gspread starts its indices at 1
+            for i, row in enumerate(ws.get_all_values(), start=1):
+                if i in range(start_row, end_row + 1):
+                    start_col_index = start_col - 1
+                    yield tuple(row[start_col_index:end_col])
+        else:
+            # no range specified, so return all the rows
+            for row in ws.get_all_values():
+                yield tuple(row)
+
+
+def togsheet(tbl, filename, credentials, worksheet_title=None,
+             share_emails=[], role='writer'):
+    """
+    Write a table to a new google sheet.
+
+    `filename` will be the title of the workbook when uploaded to google sheets.
+
+    `credentials` are used to authenticate with the google apis.
+    For more info, visit: http://gspread.readthedocs.io/en/latest/oauth2.html
+
+    If `worksheet_title` is specified, the first worksheet in the spreadsheet
+    will be renamed to the value of `worksheet_title`.
+
+    The spreadsheet will be shared with all emails in `share_emails` with
+    `role` permissions granted.
+    For more info, visit: https://developers.google.com/drive/v3/web/manage-sharing
+
+    Note: necessary scope for using togsheet is:
+         'https://spreadsheets.google.com/feeds'
+         'https://www.googleapis.com/auth/drive'
+
+    Example usage::
+        >>> import petl as etl
+        >>> from oauth2client.service_account import ServiceAccountCredentials
+        >>> scope = ['https://spreadsheets.google.com/feeds',
+                     'https://www.googleapis.com/auth/drive']
+        >>> credentials = ServiceAccountCredentials.from_json_keyfile_name('path/to/credentials.json', scope)
+        >>> cols = [[0, 1, 2], ['a', 'b', 'c']]
+        >>> tbl = etl.fromcolumns(cols)
+        >>> etl.togsheet(tbl, 'example', credentials)
+    """
+
+    import gspread
+    gspread_client = gspread.authorize(credentials)
+    spreadsheet = gspread_client.create(filename)
+    rows = len(tbl)
+    # get the max length and add [0] to take care of empty iterables
+    cols = max([0] + [len(row) for row in tbl])
+    # output to first sheet
+    worksheet = spreadsheet.sheet1
+    # match row and column length
+    worksheet.resize(rows=rows, cols=cols)
+    # rename sheet if set
+    if worksheet_title:
+        worksheet.update_title(title=worksheet_title)
+    # gspread indices start at 1, therefore row/col index starts at 1
+    for x, row in enumerate(tbl, start=1):
+        for y, val in enumerate(row, start=1):
+            worksheet.update_cell(x, y, val)
```

Comment:
The one problem about `clear` is that it will only clear the data in the spreadsheet. I was originally using `append_row` to append rows to the sheet, but the default size of a spreadsheet is 1000 x 1000, so the data would be in row `100x+`.
I think I could change it to resize the worksheet to 1 x 1, and then use `insert_row` to batch row insertion and resize. This would end up with the data in the table with one blank row at the bottom, but I believe that `gspread` wouldn't yield that empty row when pulling in from a spreadsheet. 

`appendgsheet` should be really easy to implement with `append_row` as well, but there might be a huge gap in data that way. It might be easier to have it `insert_row` at the index where the data stops, but that's much less efficient than appending.

Either way, I'm currently testing the `insert_row` logic, but Google Sheets has been inconsistent. I will try to have a more efficient version up shortly. 😄